Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar

```diff
@@ -293,71 +322,120 @@ if c:
+ argLoadmap = (args.last('loadmap')or args.last('mapload')).lower()
+ for maps in mapBG:
+  if argLoadmap in maps.lower():
+   foundMap = mapBG.get(maps)
+   if 'mapbg' in foundMap:
+    mapbg = foundMap.get('mapbg')
+   else:
+   	mapbg = ""
+   if 'mapsize' in foundMap:
+    mapsize = foundMap.get('mapsize')
+   if 'mapoptions' in foundMap:
+    mapoptions = foundMap.get('mapoptions')
+   else:
+   	mapoptions = ""
+   if 'walls' in foundMap:
+    walls = foundMap.get('walls')
+   else:
+   	walls = []
+   if 'objects' in foundMap:
+    objects = foundMap.get('objects')
+   else:
+   	objects = []
+   if typeof(foundMap) == 'SafeList':
+   	mapbg = foundMap[0]
+   	mapsize = foundMap[1]
+   	mapoptions = foundMap[2]
+   	objects = []
+   	walls = []
+   desc.append(f"Loaded the map {maps}")
+   neweffect = f"""{f"Size: {mapsize}" if mapsize else ""}{f" ~ Background: {mapbg}" if mapbg else ""}{f" ~ Options: {mapoptions}" if mapoptions else ""}{f' ~ Walls: {", ".join(walls)}' if walls else ""}{f' ~ Objects: {"/".join(objects)}' if objects else ""}{f" ~ View: {mapviewlocation}:{mapviewsize}" if mapviewsize and mapviewlocation else ""}""".strip(" ~")
+   if mapattach:
+    mapattach.add_effect('map', f"""-attack "||{neweffect}" """) 
+   break
+</drac2>
 <drac2>
 #This allows the user to create custom mapbg presets and save them to their mapBackgrounds uvar to be loaded using -mapload
 if args.last('newmapbg') or args.last('deletemapbg') or args.last('savemap') or args.last('deletemap'):
  #First option is to create mapbg preset
+ userMaps = load_json(get('mapBackgrounds','{}'))
  if args.last('newmapbg') or args.last('savemap'):
-  userMaps = load_json(get('mapBackgrounds','{}'))
-  for addMapbg in (args.get('newmapbg') or args.get('savemap')):
-   if ":" in addMapbg:
-     userMaps.update({addMapbg.split(":")[0]:addMapbg.split(":", 1)[1].split(',')})
-   else:
-     err("You must separate the map name from the map info with a : and url, size and options with a comma when adding `-savemap` entries.")
-   set_uvar("mapBackgrounds",dump_json(userMaps))
-   desc.append(f"Created the mapbg `{addMapbg.split(':')[0]}`")
- #Second option is to delete an already created map!map
+  mapName = (args.last('savemap') or args.last('newmapbg')).lower()
+  mapBGState = {}
+  formedMap = {mapName:mapBGState}
+  for key, value in mapinfo.items():
+   if value:
+    if mapinfo.get('background')!= None:
+     mapBGState["mapbg"] = mapinfo.get('background')
+    mapBGState["mapsize"] = mapsize
+    if mapinfo.get('options')!= None:
+     mapBGState["mapoptions"] = mapinfo.get('options')
+    if mapinfo.get('walls')!= None:
+     mapBGState["walls"] = mapinfo.get('walls').split(', ')
+    if mapinfo.get('objects')!= None:
+     mapBGState["objects"] = mapinfo.get('objects').split('/')
+  userMaps.update(formedMap)
+  set_uvar("mapBackgrounds",dump_json(userMaps))
+  desc.append(f"Created the map `{mapName}`")
+ #Second option is to delete an already created map
```

Comment:
New savemap...saves everything that is saved in mapattach..So user just ahs to run !map -savemap (mapname)